sound/soc/sof/topology.c

```diff
@@ -3626,6 +3627,11 @@ static void sof_complete(struct snd_soc_component *scomp)
 	 * IPC. It may be overwritten by alsa-mixer after booting up
 	 */
 	snd_sof_cache_kcontrol_val(scomp);
+
+	/* now that topology parsing is complete, set up routes for all PCM streams */
```

Comment:
I don't understand what this does?
Really no idea why things worked without this before, and now it's required for dynamic topology management?
Please help the rest of us with more comments on what this is used for?

Reply:
@plbossart I can enahnce the comment here but basically all we have today is a list of all widgets and all routes from topology in long lists. Combing through them to find the widgets/routes belonging to one pipeline is an expensive operation as you can tell from the commit that sets up the routes in this PR. Doing this dugin PCM open for every pipeline could result in long delays. This idea here is to break down the route info per PCM stream and store them beforehand, so all we're doing is looking up the list of widgets and routes in this pipeline and setting them up.

Reply:
makes sense, that design intent needs to be added in the commit message.